apps/src/templates/studioHomepages/JoinSection.jsx

```diff
@@ -61,10 +64,39 @@ const styles = {
 const JoinSection = React.createClass({
   propTypes: {
     enrolledInASection: React.PropTypes.bool.isRequired,
+    updateSections: React.PropTypes.func.isRequired
+  },
+
+  getInitialState() {
+    return {
+      sectionCode: ''
+    };
+  },
+
+  handleChange(event) {
+    this.setState({sectionCode: event.target.value});
+  },
+
+  handleKeyUp(event) {
+    if (event.key === 'Enter') {
+      this.joinSection();
+    } else if (event.key === 'Escape') {
+      this.setState(this.getInitialState());
+    }
   },
 
   joinSection() {
-    return;
+    const sectionCode = this.state.sectionCode;
+
+    this.setState(this.getInitialState());
+
+    $.post('/followers/create_async', {section_code: sectionCode})
+      .done(function (data) {
+        this.props.updateSections(data.sections);
+      }.bind(this))
+      .fail(function () {
+      }.bind(this)
+    );
```

Comment:
Use arrow functions, and you no longer need to bind(this). Also, it looks like your fail is a noop?
```
$.post('/followers/create_async', {section_code: sectionCode})
  .done(data => this.props.updateSections(data.sections))
  .fail(() => console.log('do something here if you want'))
```